onnxscript/converter_test.py

```diff
@@ -581,6 +581,19 @@ def duplicate_output(X):
         outputs = duplicate_output.to_function_proto().output
         self.assertNotEqual(outputs[0], outputs[1])
 
+    def test_bool_op_name_generation(self):
+        """Verify that python variable name is used as onnx variable name (when possible)
+        for boolean operation translation.
+        """
+
+        @script(default_opset=op)
+        def bool_op(X, Y):
+            T = X and Y
```

Comment:
python requires `and` return a bool I think. Does this work in eager mode?

Reply:
Good point. It doesn't. Perhaps this whole support should be removed, since it is not that useful. The only way to support it (in eager mode) would be to transform the AST itself, and recompiling. Doesn't seem worth it.

Reply:
I will go ahead and merge the PR, so that this fixed version is in the git history. I will separately delete the support for AND operator (unless someone has any good reason to have it).

Reply:
Just a couple more thoughts: The eager-mode does support it in the scalar case (i.e., a tensor of rank 0). Which is what is expected in if-conditions and while-conditions. But it won't handle the element-wise version for tensors (such as the condition in a Where op). So, there are some situations where it makes sense and can be supported. But not sure it is worth having the complications by supporting it. The other difference is the short-circuit evaluation, which seems to be the reason that Python doesn't support user-classes to override these boolean operators via some magic __method__. ONNX doesn't have any short-circuit evaluation (though a backend can choose to implement it that way).

Reply:
I agree. Supporting special cases may end up causing more confusion.